File: app/schemas/trip.py
import uuid
from datetime import datetime
from typing import TypedDict
from pydantic import BaseModel, ConfigDict


class TripDTO(TypedDict, total=False):
    destination: str
    price: float
    rating: float
    provider: str
    origin: str
    hotel_stars: int
    flight_price: float
    hotel_price: float


class TripCreate(BaseModel):
    destination: str
    price: float
    rating: float
    provider: str


class TripRead(BaseModel):
    id: uuid.UUID
    destination: str
    price: float
    rating: float
    provider: str
    created_at: datetime
    origin: str | None = None
    hotel_stars: int | None = None
    flight_price: float | None = None
    hotel_price: float | None = None

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/trip.ts
export interface TripRead {
  id: string;
  destination: string;
  price: number;
  rating: number;
  provider: string;
  created_at: string;
  origin: string | null;
  hotel_stars: number | null;
  flight_price: number | null;
  hotel_price: number | null;
  check_in: string | null;
  check_out: string | null;
  from_airport: string | null;
  to_airport: string | null;
  booking_url: string | null;
}

export interface SearchAIResponse {
  destination: string;
  origin: string | null;
  budget: number | null;
  results: TripRead[];
}

export interface SearchParams {
  destination: string;
  origin: string;
  check_in: string;
  check_out: string;
  max_price?: number;
  hotel_stars?: number;
}

export interface ExtractionParams {
  destination: string | null;
  origin: string | null;
  check_in: string | null;
  check_out: string | null;
  budget: number | null;
}
